nexus/reconfigurator/planning/src/planner.rs

```diff
@@ -1572,11 +1580,40 @@ impl<'a> Planner<'a> {
     /// case, we have no choice but to reconcile with the fact that the zone is
     /// now gone.
     fn can_zone_be_shut_down_safely(&self, zone: &BlueprintZoneConfig) -> bool {
-        // TODO-cleanup remove this `allow` once we populate a variant below
-        #[allow(clippy::match_single_binding)]
         match zone.zone_type.kind() {
-            // <https://github.com/oxidecomputer/omicron/issues/6404>
-            // ZoneKind::CockroachDb => todo!("check cluster status in inventory"),
+            ZoneKind::CockroachDb => {
+                debug!(self.log, "Checking if Cockroach node can shut down");
```

Comment:
I'd really like to return more structured "why we cannot update" results here, but using debug messages in the meantime.

This feels like a very doable follow-up; I'd also like to test we're getting sufficient coverage here.